src/Components/Input/types.ts

```diff
@@ -0,0 +1,23 @@
+import { SxProps } from "@mui/material";
+
+export interface InputProps {
+  inputValue: string;
+  setInputValue: (value: string) => void;
+  placeHolder?: string;
+  disabled?: boolean;
+  error?: boolean;
+  setIsPressed: (value: boolean) => void;
+  onClickIconStart?: (value: string) => void;
+  onClickIconEnd?: (value: string) => void;
+  onError?: () => void;
+  IconStart?: React.ComponentType;
+  IconEnd?: React.ComponentType;
+  IconError?: React.ComponentType;
+  showIconStart?: boolean;
+  showIconEnd?: boolean;
```

Comment:
I would have skip these 2 props, and calculate them based on the IconStart / IconEnd props ( if they are visible, then show= TRUE, else FALSE.